src/main/java/com/hida/Minter.java

```diff
@@ -0,0 +1,772 @@
+package com.hida;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Random;
+import java.util.Set;
+import org.codehaus.jackson.map.ObjectMapper;
+import java.util.TreeSet;
+
+/**
+ * Minter class creates formatted ids in json using a variety of parameters.
+ *
+ * @author Brittany Cruz
+ * @author lruffin
+ */
+public class Minter {
+
+    /**
+     * Used to add and check ids against database.
+     */
+    private final DatabaseManager DatabaseManager;
+
+    /**
+     * The mapping used to describe range of possible characters at each of the
+     * id's root's digits. There are total of 5 different ranges that a charMap
+     * can contain:
+     *
+     * <pre>
+     * d: digits only
+     * l: lower case letters only
+     * u: upper case letters only
+     * m: letters only
+     * e: any valid character specified by d, l, u, and m.
+     * </pre>
+     *
+     *
+     */
+    private String CharMap;
+
+    /**
+     * Designates what characters are contained in the id's root. There are 7
+     * types of token maps, each describing a range of possible characters in
+     * the id. This range is further affected by the variable SansVowel.
+     *
+     * <pre>
+     * DIGIT: Digit values only.
+     * LOWERCASE: Lowercase letters only.
+     * UPPERCASE: Uppercase letters only.
+     * MIXEDCASE: Lowercase and Uppercase letters only.
+     * LOWER_EXTENDED: Digit values and Lowercase letters only.
+     * UPPER_EXTENDED: Digits and Uppercase letters only
+     * MIXED_EXTENDED: All characters specified by previous tokens
+     * </pre>
+     *
+     */
+    private String TokenType;
+
+    /**
+     * Contains the mappings for either tokens or the charMaps. The AutoMinter
+     * constructor will assign token mappings. The CustomMinter constructor will
+     * assign character mappings. The values will depend on whether or sansVowel
+     * specified in the constructor.
+     */
+    private final HashMap<String, String> BaseMap = new HashMap<String, String>();

```

Comment:
nit: you don't need the type parameters on the right hand side.  This is equivalent:

```java
private final HashMap<String, String> BaseMap = new HashMap<>();
```